### PR TITLE
Add documentation for compute instance metadata

### PR DESCRIPTION
**Pull request by alexng-canuck**
Clarify which metadata keys are supported for provisioning compute instances.

### CODE REVIEW

File: docs/resources/core/instance.md [hunk 1]
```diff
@@ -55,7 +55,9 @@ The following arguments are supported:
 * `display_name` - (Optional) A user-friendly name. Does not have to be unique, and it's changeable. Avoid entering confidential information.
 * `image` - (Required) The OCID of the image used to boot the instance.
 * `ipxe_script` - (Optional) This is an advanced option. See the [instance API reference](https://docs.us-phoenix-1.oraclecloud.com/api/#/en/iaas/20160918/Instance/) for details.
-* `metadata` - (Optional) Custom metadata key/value pairs that you provide, such as the SSH public key required to connect to the instance.
+* `metadata` - (Optional) Custom metadata key/value pairs that you provide. Some possible key/value pairs:
+  * `ssh_authorized_keys` - SSH public key required to connect to the instance
```
Comment (AlaaShaker) [inline docs/resources/core/instance.md]: nit: Missing period at the end of the line for consistency.

File: docs/resources/core/instance.md [hunk 2]
```diff
@@ -55,7 +55,9 @@ The following arguments are supported:
...
+  * `user_data` - Specify cloud-init data for Linux instances. On Windows instances, this data isn't used.
```
Comment (briangustafson) [inline docs/resources/core/instance.md]: There's a bit more info at https://docs.us-phoenix-1.oraclecloud.com/api/#/en/iaas/20160918/requests/LaunchInstanceDetails - would be good to at least mention the base64 encoding, and perhaps include that in the example above. Capitalize Cloud-Init

File: docs/resources/core/instance.md (hunk 1 above)
Comment (alexng-canuck) [reply docs/resources/core/instance.md]: Fixed.

File: docs/resources/core/instance.md (hunk 2 above)
Comment (alexng-canuck) [reply docs/resources/core/instance.md]: Addressed this. Also added link to the cloud-init documentation.

**Pull request merged**